### PR TITLE
CALL FUNCTION 'SEO_CLIF_EXISTENCE_CHECK'

### LINKED ISSUE #40: CALL FUNCTION 'SEO_CLIF_EXISTENCE_CHECK'
suggest using CL_OO_CLASSNAME_SERVICE=>GET_CLASS_CATEGORY instead

no function module calls, its 2022

### PR DESCRIPTION
**Pull request by albertmink**
closes #40

### CODE REVIEW

**Comment by larshp** [comment]
CL_OO_CLASSNAME_SERVICE is the class name, not function module 😄

File: src/zcl_aff_writer.clas.abap [hunk 1]
```diff
@@ -517,16 +517,15 @@ CLASS zcl_aff_writer IMPLEMENTATION.
...
 
   METHOD get_constant_as_struc.
-    DATA clstype TYPE seoclstype.
-    CALL FUNCTION 'SEO_CLIF_EXISTENCE_CHECK'
+    cl_oo_classname_service=>get_class_category(
       EXPORTING
-        cifkey        = CONV seoclskey( name_of_source )
-      IMPORTING
-        clstype       = clstype
+        clsname            = CONV seoclsname( name_of_source )
```
Comment (larshp) [inline src/zcl_aff_writer.clas.abap:522]: I think this will also work?

```suggestion
        clsname            = CONV #( name_of_source )
```

**Comment by albertmink** [comment]
@larshp what is the abaplint observation telling me? It is not an error, but I am curious what it is about.

**Comment by larshp** [comment]
`CL_OO_CLASSNAME_SERVICE` is not used anywhere else in this repository, this PR introduces a new dependency to this class. Which is expected, its interesting for reviewers to know when new stuff from the standard system is required. Eventually this repository will have everything strictly checked, but until then this class is a void type and unknown to the linter as such.

**Review by larshp — DISMISSED** [review]

File: src/zcl_aff_writer.clas.abap [hunk 2]
```diff
@@ -517,16 +517,15 @@ CLASS zcl_aff_writer IMPLEMENTATION.
...
+        clsname            = CONV #( name_of_source )
+      RECEIVING
+        result             = DATA(clstype)
       EXCEPTIONS
-        not_specified = 1
-        not_existing  = 2
-        OTHERS        = 3.
+        class_not_existing = 1
+        OTHERS             = 2 ).
+
```
Comment (larshp) [inline src/zcl_aff_writer.clas.abap:528]: hmm, I'm not sure if this works

later in the code is:
`IF clstype = seoc_clstype_interface.`
but the result of the method call is a SEOCATEGRY, which looks different

**Comment by larshp** [comment]
Doing something like this, would be better? as it only uses RTTI,

```abap
DATA(constant_descr) = cl_abap_typedescr=>describe_by_name( name_of_source ).
IF constant_descr-TYPE_KIND = cl_abap_typedescr=>KIND_INTF.

ELSE IF constant_descr-TYPE_KIND = cl_abap_typedescr=>KIND_CLASS.

ELSE.
* not found or not relevant
ENDIF
```

**Comment by larshp** [comment]
https://github.com/open-abap/open-abap/pull/303 will fix the KIND_INTF in a bit

**Comment by larshp** [comment]
nice 👍 all RTTI, one less function module call

**Review by larshp — APPROVED** [review]
LGTM

**Comment by albertmink** [comment]
thanks for the suggestion. This in an improvement!

**Pull request merged**